compiler/quilt/tools/build.py

```diff
@@ -85,36 +89,47 @@ def _build_node(build_dir, package, name, node, fmt, target='pandas', checks_con
     Parameters
     ----------
     ancestor_args : dict
-      Any key:value pairs inherited from an ancestor. Users can this define kwargs
-      that affect entire subtrees (e.g. transform: csv for 500 .txt files)
-      Position of definition in file also matters and allows for composition
+      any transform inherited from an ancestor
+      plus any inherited handler kwargs
+      Users can thus define kwargs that affect entire subtrees
+      (e.g. transform: csv for 500 .txt files)
       and overriding of ancestor or peer values.
+      Child transform or kwargs override ancestor k:v pairs.
     """
     if _is_internal_node(node):
```

Comment:
I think logically, empty groups should fall into this category. That is, empty shouldn't be a special case.

Reply:
OK, suppose we do that. And now groups are handled by the `package.save_file` codepath? I thought about that and it also looked smelly because I would have to make a bunch of arguments that should be required for actual files (e.g. `srcfile`, `[filehash`) in `save_file` optional and that would undermine some necessary argument checks? Or am I missing something and there is an elegant way to reuse `save_file` for this?

Reply:
Per discussion with Dima: propose keeping logic as written so that internal nodes are simply descended whereas leaf nodes modify the package structure